tools/makedeb/src/System/MakeDeb.hs

```diff
@@ -86,15 +84,15 @@ makeDeb opts = shelly . silently . withTmpDir $ \tmp -> do
 
 package :: MakeDebOpts -> Sh ()
 package MakeDebOpts {..} = do
-    let f = name <> "_" <> version <> "+" <> build <> "_" <> arch
+    let f = name <> "_" <> version <> plusBuild build <> "_" <> arch
     cmd "dpkg-deb" "-b" deb (out </> fromText f <.> "deb")
 
 substitute :: MakeDebOpts -> Sh ()
 substitute MakeDebOpts {..} = flip traverseFiles (encodeString deb) $ \fname -> do
     mime <- cmd "file" "--brief" "--mime" (decodeString fname)
     when ("text/plain" `Text.isPrefixOf` mime) $
         replace [ ("<<VERSION_NUMBER>>", version)
-                , ("<<BUILD_NUMBER>>", build)
+                , ("+<<BUILD_NUMBER>>", plusBuild build)
```

Comment:
That makes sense. One could have a `<<VERSION_STRING>>` placeholder instead of the `VERSION_NUMBER` and `BUILD_NUMBER` ones, where VERSION_STRING would be replaced with the passed `version_number+build_number` or, if `build_number` is not passed, with `version_number` only. 
That would be a breaking change however and require updating all debian control files. As we eventually intend to move away from debian packages, this work might not be necessary.